evaluator/evaluate.go

```diff
@@ -165,27 +181,84 @@ func evaluateWithVmaas(ctx context.Context, tx *gorm.DB, updatesReq *vmaas.Updat
 		evaluationCnt.WithLabelValues("error-database-commit").Inc()
 		return nil, errors.New("database commit failed")
 	}
-	return vmaasData, nil
+	return updatesData, nil
 }
 
-func tryGetVmaasRequest(tx *gorm.DB, accountID int, inventoryID string,
-	requested *base.Rfc3339Timestamp) (*vmaas.UpdatesV3Request, *models.SystemPlatform, error) {
-	system := tryGetSystem(tx, accountID, inventoryID, requested)
-	if system == nil {
-		return nil, nil, nil
+func getUpdatesData(ctx context.Context, tx *gorm.DB, system *models.SystemPlatform) (*vmaas.UpdatesV2Response, error) {
+	var yumUpdates *vmaas.UpdatesV2Response
+	var yumErr error
+	if enableYumUpdatesEval {
+		yumUpdates, yumErr = tryGetYumUpdates(system)
+		if yumErr != nil {
+			// ignore broken yum updates
+			utils.Log("Can't parse yum_updates").Warn(yumErr.Error())
+		}
+
+		// in evaluator-upload return just return YumUpdates
+		if yumUpdates != nil && evalTopic == uploadTopic {
+			return yumUpdates, nil
+		}
```

Comment:
Just for clarification.
- evaluator-upload should evaluate systems only by yum_updates and it should not call vmaas.
- evaluator-recalc should use yum_updates and vmaas response and merge updates
is that correct?

Reply:
Exactly.